zork.js

```diff
@@ -0,0 +1,189 @@
+const readline = require('readline');
+const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+
+function ask(questionText) {
+    return new Promise((resolve, reject) => {
+        readlineInterface.question(questionText, resolve);
+    });
+}
+
+// remember the StateMachine lecture
+// https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
+// let states = {
+//   'roomOne': { canChangeTo: [ 'roomTwo' ] },
+//   'roomTwo': { canChangeTo: [ 'roomThree' ] },
+//   'roomThree': { canChangeTo: [ 'roomOne' ] }
+// };
+
+// let currentState = "green";
+
+// function enterState(newState) {
+//   let validTransitions = states[currentState].canChangeTo;
+//   if (validTransitions.includes(newState)) {
+//     currentState = newState;
+//   } else {
+//     throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
+//   }
+// }
+
+let medicalSupplies
+let food
+let sign
+let keyPad
+let currentRoom
+let door2
+let door3
+let medSupplyRoom
+
+
+
+
+
+
+class Room {
+    constructor(roomName, roomDescription, items, puzzles, connectingRooms) {
+        this.roomName = roomName,
+            this.roomDescription = roomDescription,
+            this.items = items,
+            this.puzzles = puzzles,
+            this.connectingRooms = connectingRooms,
```

Comment:
The way your room connections are set up it will probably make more sense to have the person move into a room by name rather than direction. If you decide to go this route you might want to tell the player which rooms connect in the room description.